tiny-firmware/firmware/fsm_skycoin_impl.c

```diff
@@ -107,7 +107,9 @@ ErrCode_t msgSkycoinSignMessageImpl(SkycoinSignMessage *msg, ResponseSkycoinSign
     uint8_t seckey[SKYCOIN_SECKEY_LEN] = {0};
     uint8_t digest[SHA256_DIGEST_LENGTH] = {0};
     uint8_t signature[SKYCOIN_SIG_LEN];
-    if (fsm_getKeyPairAtIndex(1, pubkey, seckey, NULL, msg->address_n) != ErrOk) {
+    int (*address_from_pubkey)(const uint8_t*, char*, size_t*);
+    address_from_pubkey = &skycoin_address_from_pubkey;
+    if (fsm_getKeyPairAtIndex(1, pubkey, seckey, NULL, msg->address_n, address_from_pubkey) != ErrOk) {
```

Comment:
You can omit creating `address_from_pubkey` variable and directly path `&skycoin_address_from_pubkey` to fsm_getKeyPairAtIndex.